### PR TITLE
Add Builtin.assumeAlignment with support for misaligned loads

### PR DESCRIPTION
**Pull request by atrick**
Add Builtin.assumeAlignment(RawPointer, Int64)

This adds compiler support for misaligned raw pointer access. A Swift Evolution proposal is required before users can take advantage of this.

Preserve current behavior by adding assumeAlignment to UnsafeRawPointer.load(). I expect this builtin to be removed in the default case after proposing the change in Swift Evolution.

Optimize Builtin.assumeAlignment -> pointer_to_address

Case #1. Literal zero = natural alignment
   %1 = integer_literal $Builtin.Int64, 0
   %2 = builtin "uncheckedAssertAlignment"
	(%0 : $Builtin.RawPointer, %1 : $Builtin.Int64) : $Builtin.RawPointer
   %3 = pointer_to_address %2 : $Builtin.RawPointer to [align=1] $*Int

   Erases the `pointer_to_address` `[align=]` attribute:

Case #2. Literal nonzero = forced alignment.

   %1 = integer_literal $Builtin.Int64, 16
   %2 = builtin "uncheckedAssertAlignment"
	(%0 : $Builtin.RawPointer, %1 : $Builtin.Int64) : $Builtin.RawPointer
   %3 = pointer_to_address %2 : $Builtin.RawPointer to [align=1] $*Int

   Promotes the `pointer_to_address` `[align=]` attribute to a higher value.

Case #3. Folded dynamic alignment

   %1 = builtin "alignof"<T>(%0 : $@thin T.Type) : $Builtin.Word
   %2 = builtin "uncheckedAssertAlignment"
	(%0 : $Builtin.RawPointer, %1 : $Builtin.Int64) : $Builtin.RawPointer
   %3 = pointer_to_address %2 : $Builtin.RawPointer to [align=1] $*T

   Erases the `pointer_to_address` `[align=]` attribute.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci test

**Review by eeckstein — APPROVED** [review]
lgtm!

**Comment by atrick** [comment]
@swift-ci test

**Comment by atrick** [comment]
@swift-ci test

File: stdlib/public/core/UnsafeRawPointer.swift [hunk 1]
```diff
@@ -355,7 +355,13 @@ public struct UnsafeRawPointer: _Pointer, Sendable {
         & (UInt(MemoryLayout<T>.alignment) - 1)),
       "load from misaligned raw pointer")
 
-    return Builtin.loadRaw((self + offset)._rawValue)
+    let rawPointer = (self + offset)._rawValue
+
+    // TODO: to support misaligned raw loads, simply remove this assumption.
+    let alignedPointer =
+      Builtin.assumeAlignment(rawPointer,
```
Comment (CodaFi) [inline stdlib/public/core/UnsafeRawPointer.swift:362]: The use of this new builtin needs to be guarded by a feature flag.

File: stdlib/public/core/UnsafeRawPointer.swift [hunk 2]
```diff
@@ -901,7 +907,13 @@ public struct UnsafeMutableRawPointer: _Pointer, Sendable {
...
```
Comment (CodaFi) [inline stdlib/public/core/UnsafeRawPointer.swift:914]: Here too.

**Pull request merged**